spec/defines/pip_spec.rb

```diff
@@ -64,22 +64,10 @@
 
         it { is_expected.not_to contain_exec('pip_install_rpyc').with_command(%r{--proxy}) }
       end
-      context 'does not add proxy to search command if set to latest and proxy is unset' do
-        let(:params) { { ensure: 'latest' } }
-
-        it { is_expected.not_to contain_exec('pip_install_rpyc').with_command(%r{--proxy}) }
-        it { is_expected.to contain_exec('pip_install_rpyc').without_unless(%r{--proxy}) }
-      end
       context 'adds proxy to install command if proxy set' do
         let(:params) { { proxy: 'http://my.proxy:3128' } }
 
-        it { is_expected.to contain_exec('pip_install_rpyc').with_command("pip wheel --help > /dev/null 2>&1 && { pip show wheel > /dev/null 2>&1 || wheel_support_flag='--no-binary :all:'; } ; { pip --log /tmp/pip.log install $wheel_support_flag  --proxy=http://my.proxy:3128   rpyc || pip --log /tmp/pip.log install  --proxy=http://my.proxy:3128   rpyc ;}") }
-      end
-      context 'adds proxy to search command if set to latest' do
-        let(:params) { { proxy: 'http://my.proxy:3128', ensure: 'latest' } }
-
-        it { is_expected.to contain_exec('pip_install_rpyc').with_command("pip wheel --help > /dev/null 2>&1 && { pip show wheel > /dev/null 2>&1 || wheel_support_flag='--no-binary :all:'; } ; { pip --log /tmp/pip.log install --upgrade $wheel_support_flag  --proxy=http://my.proxy:3128   rpyc || pip --log /tmp/pip.log install --upgrade  --proxy=http://my.proxy:3128   rpyc ;}") }
-        it { is_expected.to contain_exec('pip_install_rpyc').with_unless('pip search  --proxy=http://my.proxy:3128 rpyc | grep -i INSTALLED.*latest') }
```

Comment:
I think this check also validated the proxy went through as defined in params.

Reply:
Oh, right. I will restore this but without checking "unless". Would that be ok, @danquack ?

Reply:
Yeah, as long as we’re not losing any validation checks.

Reply:
In the end I removed those as we don't have "search command" anymore, so there is no point in checking if it contains proxy.